python/test/unit/language/test_core.py

```diff
@@ -2966,6 +2968,45 @@ def histogram_kernel(x_ptr, z_ptr, M: tl.constexpr, N: tl.constexpr):
     assert (z_torch == z).all()
 
 
+@pytest.mark.interpreter
+def test_histogram_complex_layout(device):
```

Comment:
can you move it in `python/test/gluon/test_lowerings.py`

Reply:
even better if we can just add the layout here:
https://github.com/triton-lang/triton/blob/main/python/test/gluon/test_lowerings.py#L245

Reply:
I think this will require a new test because LinearLayout is very tied to the shape of the tensor it's associated with, and I'd need to calculate a new basis for each size :(